jenkins-scripts/dsl/test.dsl

```diff
@@ -16,3 +16,16 @@ OSRFLinuxCompilationAnyGitHub.create(ignition_ci_pr_job,
                                      false,
                                      false,
                                      ['main'])
+
```

Comment:
I'm fine with have a test job that does the same than the real job but I'm not so happy with having duplicate code for both since we probably will evolve the DSL over the time. Would be great to having a common DSL job constructor method or helper.

Reply:
Do you mean a constructor that creates both test and real ones?

Reply:
yes, a helper function to construct both without having to duplicate code.

Reply:
After a208ead, I couldn't find a way to link the function between both dsl files

Reply:
If I add it to Globals or other script, job function is not recognized.  @jrivero do you have any ideas to workaround this?

Reply:
> If I add it to Globals or other script, job function is not recognized. @j-rivero do you have any ideas to workaround this?

That is probably because the files have the `dsl` extension and not `groovy`. If you do something like:
```diff
diff --git a/jenkins-scripts/dsl/core.dsl b/jenkins-scripts/dsl/core.dsl
  index 8422a27b..3ff14dc4 100644
  --- a/jenkins-scripts/dsl/core.dsl
  +++ b/jenkins-scripts/dsl/core.dsl
  @@ -5,6 +5,10 @@ def NO_TESTING = false
   def NO_BRANCHES = []
   def NO_GITHUB_PR_INTEGRATION = false
   
  +static foo() {
  +  println "lala"
  +}
  +
   def update_vcpkg_snapshot_job = job("_vcpkg_update_snapshot")
   OSRFWinBase.create(update_vcpkg_snapshot_job)
   update_vcpkg_snapshot_job.with
  diff --git a/jenkins-scripts/dsl/test.dsl b/jenkins-scripts/dsl/test.dsl
  index 151f2d20..56a79e5f 100644
  --- a/jenkins-scripts/dsl/test.dsl
  +++ b/jenkins-scripts/dsl/test.dsl
  @@ -1,8 +1,14 @@
   import _configs_.*
   import javaposse.jobdsl.dsl.Job
   
   Globals.default_emails = "jrivero@osrfoundation.org"
   
  +import static core.foo 
  +
  +core.foo()
  +
  +
   // Jobs
   def ignition_ci_job = job("_test_job_from_dsl")
   OSRFLinuxBase.create(ignition_ci_job)

```
wont' work but if after that you do: `ln -s core.dsl core.groovy` Groovy will probably find the class.

Reply:
Implementation of your suggestion: 93c0ebb

I'm getting: 

```
Caused by: javaposse.jobdsl.dsl.DslScriptException: (core.groovy, line 13) No signature of method: static core.job() is applicable for argument types: (java.lang.String) values: [outdated_job_runner]
```

I'm suspecting that I can't use job creation if it isn't in a dsl script, not sure

Reply:
> I'm suspecting that I can't use job creation if it isn't in a dsl script, not sure

I don't think that this has something to do. Don't worry, leave the code duplicated by now, I'll try to change it with a better solution than the symlink.

Reply:
Thanks! Already changed it in 140b57c